### PR TITLE
Add integration test case for invalid credentials

### PR DESCRIPTION
**Pull request by timebertt**
**How to categorize this PR?**

/area usability
/kind test
/platform aws

**What this PR does / why we need it**:

This PR tests https://github.com/gardener/terraformer/pull/93 and https://github.com/gardener/gardener/pull/3950 by adding an integration test case for invalid credentials.
It creates an infrastructure resource but uses invalid credentials and ensures that the `ERR_INFRA_UNAUTHORIZED` is correctly detected from the termination message and that the infrastructure can still be deleted afterwards (as it's a no-op).

**Which issue(s) this PR fixes**:

Ref https://github.com/gardener/terraformer/issues/67

**Special notes for your reviewer**:

**Release note**:

```other operator
NONE
```

### CODE REVIEW

**Comment by timebertt** [comment]
Dropped `replace g/g` and terraformer dev version.
/ready
/test

File: test/integration/infrastructure/infrastructure_test.go [hunk 1]
```diff
@@ -250,6 +253,102 @@ var _ = Describe("Infrastructure tests", func() {
...
+				infra     *extensionsv1alpha1.Infrastructure
+			)
+
+			framework.AddCleanupAction(func() {
+				By("cleaning up namespace and cluster")
+				Expect(client.IgnoreNotFound(c.Delete(ctx, namespace))).To(Succeed())
+				Expect(client.IgnoreNotFound(c.Delete(ctx, cluster))).To(Succeed())
+			})
+
+			defer func() {
```
Comment (ialidzhikov) [inline test/integration/infrastructure/infrastructure_test.go:278]: `defer` won't be called if the test execution is interrupted by SIGINT/SIGTERM. You could use `AddCleanupAction` instead to make sure that the test won't leak any infra if it is interrupted.

Comment (timebertt) [reply test/integration/infrastructure/infrastructure_test.go:278]: Well, in this test we can be sure, that no infra is leaked as we pass invalid credentials.

As for the other test cases, we should actually also move the deletion logic to a `defer` and only do actual cleanup on AWS in a `CleanupAction` (see https://github.com/gardener/terraformer/pull/57 for more background on that).
However, I didn't tackle this topic in this PR.

Comment (ialidzhikov) [reply test/integration/infrastructure/infrastructure_test.go:278]: > Well, in this test we can be sure, that no infra is leaked as we pass invalid credentials.

Ah, yep, I missed this point.

**Review by ialidzhikov — APPROVED** [review]
/lgtm

**Review by rfranzke — APPROVED** [review]

**Pull request merged**